Lib/test/test_codeccallbacks.py

```diff
@@ -1235,6 +1236,37 @@ class FakeUnicodeError(Exception):
                     with self.assertRaises((TypeError, FakeUnicodeError)):
                         handler(FakeUnicodeError())
 
+    def test_reject_unregister_builtin_error_policy(self):
+        for policy in [
+            'strict', 'ignore', 'replace', 'backslashreplace', 'namereplace',
+            'xmlcharrefreplace', 'surrogateescape', 'surrogatepass',
+        ]:
+            with self.subTest(policy):
+                self.assertRaises(ValueError, _codecs_unregister_error, policy)
+
+    def test_unregister_custom_error_policy(self):
+        def custom_handler(exc):
+            raise exc
+
+        # We want a unique ID in case the test is executed multiple times
+        # to be sure that we always try to un-register a new error policy.
+        unique_id = random.getrandbits(128)
+        custom_name = f'test.test_unregister_error.custom.{unique_id}'
+        self.assertRaises(LookupError, codecs.lookup_error, custom_name)
```

Comment:
I don't get why this code has to be so complicated since you unregister the handler at each run.

```suggestion
        # We want a unique ID in case the test is executed multiple times
        # to be sure that we always try to un-register a new error policy.
        unique_id = random.getrandbits(128)
        custom_name = f'test.test_unregister_error.custom.{unique_id}'
        self.assertRaises(LookupError, codecs.lookup_error, custom_name)
```

Why not just:

```py
custom_name = 'test_unregister_error'
```

Reply:
I wanted to be sure that in the event the test failed for some obscure reason and if it's repeated (e.g., when using `-R :`), then the test is always executed in a "fresh" setup (in the end, there will be a failure, but at least it won't be due to the error policy being already there)

Reply:
By observing the other tests, I see that they don't care about this, so I'll just do the same on my side.